ivadomed/utils.py

```diff
@@ -620,6 +640,115 @@ def threshold_predictions(predictions, thr=0.5):
     return thresholded_preds
 
 
+def segment_volume(fname_model, fname_model_metadata, fname_image, fname_roi=None):
+    """Segment an image.
+
+    Segment an image (fname_image) using a already trained model (fname_model) given its
+    training parameters (fname_model_metadata). If provided, a RegionOfInterest (fname_roi)
+    is used to crop the image prior to segment it.
+
+    Args:
+        fname_model (string): model filename (.pt) to use.
+        fname_model_metadata (string): Configuration file filename (.json) used for the training,
+            see https://github.com/neuropoly/ivado-medical-imaging/wiki/configuration-file
+        fname_image (string): image filename (e.g. .nii.gz) to segment.
+        roi_fname (string): Binary image filename (e.g. .nii.gz) defining a region of interest,
+            e.g. spinal cord centerline, used to crop the image prior to segment it if provided.
+    Returns:
+        nibabelObject: Object containing the soft segmentation.
+
+    """
+    # Define device
+    device = torch.device("cpu")
+
+    # Load model training config
+    with open(fname_model_metadata, "r") as fhandle:
+        context = json.load(fhandle)
+
+    # If ROI is not provided then force center cropping
+    if fname_roi is None and 'ROICrop2D' in context["transformation_validation"].keys():
+        context["transformation_validation"] = dict((key, value) if key != 'ROICrop2D'
+                                                    else ('CenterCrop2D', value)
+                                                    for (key, value) in context["transformation_validation"].items())
+
+    # Compose transforms
+    do_transforms = compose_transforms(context['transformation_validation'])
+
+    # Undo Transforms
+    undo_transforms = ivadomed_transforms.UndoCompose(do_transforms)
+
+    # Load data
+    filename_pairs = [([fname_image], fname_image, fname_roi, [{}])]
```

Comment:
The second element in the list represents the gt, since this function is only for prediction it should be None (the gt is not provided). Which also implicates the parameter `filter_empty_mask` should be at `False` in config files, since the data is unlabeled.

Reply:
I put `None` initially but it brought some bugs in `medicaltorch` --> easy and lazy fix: send `fname_image`.
`filter_empty_mask` --> maybe I should force this parameter to `False` in this function to avoid this.

Reply:
I put `None` in `filename_pairs` as suggested but it requires me to add some `if self.labeled` in the transforms. And to force it to `False` (since set to `True` by default).

Reply:
Note that we need this [PR](https://github.com/neuropoly/medicaltorch/pull/18) to be merged to ensure good behaviour here in `ivadomed`.